expression/builtin_cast_vec_test.go

```diff
@@ -24,6 +24,7 @@ import (
 var vecBuiltinCastCases = map[string][]vecExprBenchCase{
 	ast.Cast: {
 		{retEvalType: types.ETInt, childrenTypes: []types.EvalType{types.ETInt}},
+		{retEvalType: types.ETInt, childrenTypes: []types.EvalType{types.ETDecimal}},
```

Comment:
It seems that the `defaultGenerator` will generate an integer for `ETDecimal`, thus the `ModeHalfEven` round can not be tested. We need to refine the behavior of `defaultGenerator` or define a new data generator.

Reply:
I think I can add `childrenFieldTypes: []*types.FieldType{types.NewFieldType(mysql.TypeNewDecimal)}` in test case ?

Reply:
We can refine the `defaultGener.gen`.
An example:

``` go
	case types.ETDecimal:
		d := new(types.MyDecimal)
		f := float64(rand.Int63())
		if rand.Float64() < 0.5 {
			f *= -1
		}
		if err := d.FromFloat64(f/10); err != nil {
			panic(err)
		}
```
You can make the generator more random and effective.

Reply:
But I don't know why the `defaultGenerator` will generate an integer for `ETDecimal`?
```
if err := d.FromFloat64(f/10); err != nil {
    panic(err)
}
```
I print `d.String()` and I found it was decimal.

Reply:
https://github.com/pingcap/tidb/blob/master/expression/bench_test.go#L226-L231

Reply:
Sorry, my fault here. I misunderstood it.